src/world/world.cpp

```diff
@@ -154,8 +154,8 @@ void
 World::update_marked_chunks_mesh() {
     for (auto chunk_pos : chunks_to_update_) {
         GlobalContext& context = GlobalContext::instance();
-        context.submit(
-            [this](ChunkIndex p) { this->update_single_mesh(p); }, chunk_pos
+        context.submit_task(
+            [this, chunk_pos]() { this->update_single_mesh(chunk_pos); }
```

Comment:
is `update_single_mesh` thread safe?

Reply:
no this would require major changes. will create an issue

Reply:
#83